component/service/src/test/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesTest.java

```
@@ -73,6 +73,68 @@ public void tearDown() throws Exception {
     removeResource(spaceRestResources.getClass());
   }
 
+  public void testFillSpaceFromModel () throws Exception {
+    startSessionAs("root");
+    List listOfResponses = new ArrayList<ContainerResponse>();
+    /*
+    *
+    * Test of 'private' field with a mix of upper/lower cases
+    */
+    String input = "{\"visibility\":PRIVATE}";
+    Space space = getSpaceInstance(1, "root");
+
+    ContainerResponse response = getResponse("PUT", getURLResource("spaces/" + space.getId()), input);
+    listOfResponses.add(response);
+
+    input = "{\"visibility\":private}";
+
+    response = getResponse("PUT", getURLResource("spaces/" + space.getId()), input);
+    listOfResponses.add(response);
+
+    input = "{\"visibility\":PriVatE}";
+
+    response = getResponse("PUT", getURLResource("spaces/" + space.getId()), input);
+    listOfResponses.add(response);
+
+    for(Object res : listOfResponses){
```

Ok so I would suggest you to include the getResponse part in the loop too, put the inputs and the expected output in a map, iterate over it and for each value call the getResponse and check the result. This will avoid sorting results, reusing code and also failing fast.
Actually this is Parameterized tests of JUnit but this feature sucks in JUnit, it is not flexible enough.